### PR TITLE
pitch_set_default should revert to linear scaler if keylock is off

### PR DESCRIPTION
**Pull request by ywwg**
If keylock is off, and the rate slider has been moved, pitch_set_default should cause the engine to go back to the linear scaler.  Otherwise if the user does a pitch change and then tries to reset it, they are stuck with the keylock engine until they load a new track.

https://bugs.launchpad.net/mixxx/+bug/1435124

### CODE REVIEW

File: src/engine/keycontrol.cpp [hunk 1]
```diff
@@ -287,18 +287,26 @@ void KeyControl::slotPitchChanged(double pitch) {
 
 void KeyControl::updatePitch() {
     double pitch = m_pPitch->get();
+    double pitchKnobRatio = KeyUtils::semitoneChangeToPowerOf2(pitch);
```
Comment (daschuer) [inline src/engine/keycontrol.cpp]: move closer to usage

Comment (ywwg) [inline src/engine/keycontrol.cpp]: done -- pitchknobratio is used inside and outside the if so it can't be a lot closer

**Comment by daschuer** [comment]
I am afraid this PR messes to much with the internal values we have, without updating the COs.
It is hard to test, because we have no skin that displays all controls. The developer menu helps a bit. 

pitch_set_default sets the pitch back to the natural track pitch. This cannot be done by the linear scaler if the speed slider is off center. So it looks like the desired target of this PR cannot be achieved. 

You want probably pitch_adjust_set_default. This should enable the linear scaler in any case. 
But it looks like we have an issue here as well. I have just checked with master build it and it should go back to the linear pitch but the pitch display also resets to 0.

**Comment by ywwg** [comment]
that's why I asked the original question in the bug.  What CO should I be setting to disable pitch adjustment and go back to the linear scalar?  There should be a clear answer and I don't understand why there are so many different COs.

if pitch_adjust_set_default is the correct CO, and it's not working, then please help me write a PR to fix that.

**Comment by daschuer** [comment]
It was only a display issue. Interesting that no one has noticed it before. 
https://github.com/mixxxdj/mixxx/commit/68aa11029fa540c503430866f3668ba8c978a09e
pitch_adjust_set_default should now work for your purpose to go back to linear pitch, without any additional changes.

**Comment by ywwg** [comment]
ok I'll update the skin

**Pull request closed without merging**